samples: matter: Added DSLS support to the smoke co alarm (#19151)
nrfconnect/sdk-nrf: samples/matter/smoke_co_alarm/README.rst

* End of service alert - Issued if the device service was ended either by the expiration date or other physical conditions, and it needs to be replaced.
* Battery level alert - Issued if the device battery level is too low.

The sample implements two instances of a Power Source cluster:

* Wired power source on the endpoint 0

> * Batter power source on the endpoint 1

Reviewer: ```suggestion
* Battery power source on the endpoint 1
```

The device starts operation in the Short Idle Time (SIT) mode and remains in it until it is commissioned to the Matter fabric and registers the first ICD client.
It then switches the operation mode to LIT to reduce the power consumption.

The sample supports ICD Dynamic SIT LIT switching (DSLS) feature to switch between SIT and LIT modes, depending on the used power source.

> The device uses a SIT mode, when the wired power source is active.

Reviewer: ```suggestion
The device uses the SIT mode, when the wired power source is active.
```

> Otherwise it moves to the LIT mode, to present that it is possible to save the energy, when using battery power source.

Reviewer: ```suggestion
Otherwise, it switches to the LIT mode, to indicate that it is possible to save the energy, when using a battery power source.
```

> The dynamic switching is possible only, if the ICD device meets all criteria for operating in the LIT mode (it has at least one client registered).

Reviewer: ```suggestion
Dynamic switching is possible only if the ICD device meets all criteria for operating in the LIT mode (it has at least one client registered).
```

> The DSLS support is by default enabled.

Reviewer: ```suggestion
DSLS support is enabled by default.
```
